ocaml/xapi/xapi_vm.ml

```diff
@@ -238,6 +238,39 @@ let assert_memory_constraints ~__context ~vm platformdata =
    same VM twice during memory calculations to determine whether a given VM can start on a particular host..
 *)
 
+let update_platform_secureboot ~__context ~self platform =
+  match List.assoc "secureboot" platform with
+  | exception Not_found -> platform
+  | "auto" -> ("secureboot", string_of_bool (Db.Pool.get_uefi_certificates ~__context ~self <> "")) :: (List.remove_assoc "secureboot" platform)
+  | _ -> platform
+
+let extract_certificate_file name =
+  if String.contains name '/' then
+	(* Internal error: tarfile not created correctly *) 
+	failwith ("Invalid path in certificate tarball: " ^ name);
+  let path = Filename.concat !Xapi_globs.varstore_dir name in
+  Helpers.touch_file path;
+  path
+
+let with_temp_file_contents ~contents f =
+  let filename, out = Filename.open_temp_file "xapi_uefi_certificates" "tar" in
+  Xapi_stdext_pervasives.Pervasiveext.finally (fun () ->
+    Xapi_stdext_pervasives.Pervasiveext.finally
+      (fun () -> output_string out contents)
+      (fun () -> close_out out);
+    Unixext.with_file filename [Unix.O_RDONLY] 0 f
+  ) (fun () -> Sys.remove filename)
+
+let save_uefi_certificates_to_dir ~__context ~pool ~vm =
+  let uefi_key = !Xapi_globs.varstore_dir ^ "KEK.auth" in
+  if not (Sys.file_exists uefi_key) then
+    if Sys.file_exists !Xapi_globs.varstore_dir && Sys.is_directory !Xapi_globs.varstore_dir then
```

Comment:
We can keep this as is but I think

* it would have been enough to test for `is_directory`
* if the content is empty, a log line would probably be helpful - I assume this should not happen.